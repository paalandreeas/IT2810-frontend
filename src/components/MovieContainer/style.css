:root {
  --width: 200px;
  --border-radius: 15px;
}

.result {
  width: var(--width);
  min-width: var(--width);
  border-radius: var(--border-radius);
  overflow: hidden;
  cursor: pointer;
  transition-property: opacity;
  transition-duration: 1s;
}

.resultImg {
  width: 200px;
  border-radius: calc(var(--border-radius) * 0.7);
}

.title {
  padding: 20px;
  text-align: center;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}
