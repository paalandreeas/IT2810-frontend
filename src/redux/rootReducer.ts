import { combineReducers } from "redux";
import searchReducer from "./search/searchReducer";
import filterReducer from "./filter/filterReducer";
import sortReducer from "./sort/sortReducer";
import movieReducer from "./movie/movieReducer";
import userReducer from "./user/userReducer";
import reviewReducer from "./review/reviewReducer";
import alertReducer from "./alert/alertReducer";

// Creates a root reducer from all the reducers.
const rootReducer = combineReducers({
  search: searchReducer,
  filter: filterReducer,
  sort: sortReducer,
  movieInfo: movieReducer,
  userInfo: userReducer,
  reviewInfo: reviewReducer,
  alertInfo: alertReducer,
});

export default rootReducer;
