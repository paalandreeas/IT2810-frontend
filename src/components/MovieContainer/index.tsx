import { Box } from "@material-ui/core";
import React from "react";
import { useHistory } from "react-router-dom";
import "./style.css";

// Takes in id, title and posterPath as props from parent.
type Props = {
  id: string;
  title: string;
  posterPath: string;
};

const MovieContainer: React.FunctionComponent<Props> = (props) => {
  // The useHistory hook gives access to the history instance used for navigation.
  const history = useHistory();

  return (
    // Returns a Box (MUI) with an image and title according to the props it got from the SearchResults component.
    // When clicked, triggers openMovieDialog action with the current id from redux state.
    <Box
      // Changes the history when clicked
      onClick={() => history.push("movie/" + props.id)}
      className="result"
      bgcolor="secondary.light"
      boxShadow={3}
      color="text.secondary"
      data-testid="boxen"
    >
      <img
        className="resultImg"
        src={props.posterPath}
        alt={props.title + " poster"}
      />
      <p className="title">{props.title}</p>
    </Box>
  );
};

export default MovieContainer;
