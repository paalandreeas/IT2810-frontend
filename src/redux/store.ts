import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";
import rootReducer from "./rootReducer";

// Creates a redux store from the rootReducer and applies devTools and thunk middleware
const store = createStore(
    rootReducer,
    composeWithDevTools(applyMiddleware(thunk))
);

export default store;
