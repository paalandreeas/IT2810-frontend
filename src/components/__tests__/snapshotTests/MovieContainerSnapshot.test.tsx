import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { applyMiddleware, createStore } from "redux";
import rootReducer from "../../../redux/rootReducer";
import thunk from "redux-thunk";
import { BrowserRouter as Router } from "react-router-dom";
import MovieContainer from "../../MovieContainer";

//creates a store for the provider
const store = createStore(rootReducer, applyMiddleware(thunk));

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <MovieContainer
            id={"1"}
            title={"title1"}
            posterPath={"posterpath1"}
          />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
