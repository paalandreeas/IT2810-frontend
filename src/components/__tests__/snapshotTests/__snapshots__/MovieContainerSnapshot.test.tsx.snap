// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-1 result"
  data-testid="boxen"
  onClick={[Function]}
>
  <img
    alt="title1 poster"
    className="resultImg"
    src="posterpath1"
  />
  <p
    className="title"
  >
    title1
  </p>
</div>
`;
